api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -740,6 +742,33 @@ public void signOut(DbUser user) {
     directoryService.signOut(user.getUsername());
   }
 
+  @Override
+  public DbUserInitialCreditsExpiration createInitialCreditsExpiration(DbUser user) {
+    long initialCreditsValidityPeriodDays =
+        configProvider.get().billing.initialCreditsValidityPeriodDays;
+    Timestamp now = clockNow();
+    Timestamp expirationTime =
+        new Timestamp(now.getTime() + TimeUnit.DAYS.toMillis(initialCreditsValidityPeriodDays));
+    DbUserInitialCreditsExpiration userInitialCreditsExpiration =
+        new DbUserInitialCreditsExpiration()
+            .setCreditStartTime(now)
+            .setExpirationTime(expirationTime)
+            .setUser(user)
+            .setNotificationStatus(NotificationStatus.NO_NOTIFICATION_SENT);
+    user.setUserInitialCreditsExpiration(userInitialCreditsExpiration);
+    return userInitialCreditsExpiration;
+  }
+
+  @Override
+  public void setInitialCreditsExpirationBypassed(DbUser user, boolean isBypassed) {
+    DbUserInitialCreditsExpiration userInitialCreditsExpiration =
+        user.getUserInitialCreditsExpiration();
+    if (userInitialCreditsExpiration == null) {
```

Comment:
what I mean is that the expiration time is not relevant at bypass time, but it could become relevant if the user is unbypassed later.  Dunno how likely that would be - or really what our expected behavior should be, so I'm not requesting any changes here.